substrate/frame/ranked-collective/src/benchmarking.rs

```diff
@@ -56,131 +58,304 @@ fn make_member<T: Config<I>, I: 'static>(rank: Rank) -> T::AccountId {
+			T::Polls::create_ongoing(class.clone())
+				.expect("Poll creation should succeed for rank 0")
+		} else {
+			Default::default()
+		};
 
-		// Vote in the poll by each of `n` members
-		for i in 0..n {
-			let who = make_member::<T, I>(rank);
-			assert_ok!(Pallet::<T, I>::vote(SystemOrigin::Signed(who).into(), poll, true));
+		// Simulate voting in the poll by `n` members.
+		for _ in 0..n {
+			let voter = make_member::<T, I>(rank);
+			let result = Pallet::<T, I>::vote(SystemOrigin::Signed(voter).into(), poll, true);
+
+			// Check voting results based on class existence.
+			if class_exists {
+				assert_ok!(result);
+			} else {
+				assert_err!(result, crate::Error::<T, I>::NotPolling);
+			}
+		}
+
+		// End the poll if the class exists.
+		if class_exists {
+			T::Polls::end_ongoing(poll, false).expect("Poll should be able to end");
 		}
 
-		// End the poll.
-		T::Polls::end_ongoing(poll, false).expect("Must always be able to end a poll");
+		// Verify the number of votes cast.
+		let expected_votes = if class_exists { n as usize } else { 0 };
+		assert_eq!(Voting::<T, I>::iter_prefix(poll).count(), expected_votes);
 
-		assert_eq!(Voting::<T, I>::iter_prefix(poll).count(), n as usize);
-	}: _(SystemOrigin::Signed(whitelisted_caller()), poll, n)
-	verify {
+		// Benchmark the cleanup function.
+		#[block]
+		{
+			let _ = Pallet::<T, I>::cleanup_poll(
+				SystemOrigin::Signed(whitelisted_caller()).into(),
+				poll,
+				n,
+			);
+		}
+
+		// Ensure all votes are cleaned up.
 		assert_eq!(Voting::<T, I>::iter().count(), 0);
+		Ok(())
 	}
 
-	exchange_member {
+	#[benchmark]
+	fn exchange_member() -> Result<(), BenchmarkError> {
+		// Create an existing member.
 		let who = make_member::<T, I>(1);
 		T::BenchmarkSetup::ensure_member(&who);
 		let who_lookup = T::Lookup::unlookup(who.clone());
+
+		// Create a new account for the new member.
 		let new_who = account::<T::AccountId>("new-member", 0, SEED);
 		let new_who_lookup = T::Lookup::unlookup(new_who.clone());
+
+		// Attempt to get the successful origin for exchanging a member.
 		let origin =
 			T::ExchangeOrigin::try_successful_origin().map_err(|_| BenchmarkError::Weightless)?;
+
+		// Create the `Call` to exchange the member.
 		let call = Call::<T, I>::exchange_member { who: who_lookup, new_who: new_who_lookup };
-	}: { call.dispatch_bypass_filter(origin)? }
-	verify {
+
+		// Dispatch the call and measure execution.
+		#[block]
+		{
+			call.dispatch_bypass_filter(origin)?;
+		}
+
+		// Check that the new member was successfully exchanged and holds the correct rank.
 		assert_eq!(Members::<T, I>::get(&new_who).unwrap().rank, 1);
+
+		// Ensure the old member no longer exists.
 		assert_eq!(Members::<T, I>::get(&who), None);
-		assert_has_event::<T, I>(Event::MemberExchanged { who, new_who }.into());
+
+		// Ensure the correct event was emitted.
+		assert_has_event::<T, I>(
+			Event::MemberExchanged { who: who.clone(), new_who: new_who.clone() }.into(),
```

Comment:
```suggestion
			Event::MemberExchanged { who, new_who }.into(),
```

Or is the macro forcing you to do this cloning?